XUtils.h

```diff
@@ -111,4 +111,7 @@ int compareRealNumbers(double a, double b);
    nonnegative. */
 double sumPositiveValues(const double* array, size_t count);
 
+/* IEC unit prefixes */
+static const char unitPrefixes[] = { 'K', 'M', 'G', 'T', 'P', 'E', 'Z', 'Y', 'R', 'Q' };
```

Comment:
```c
static const char* unitPrefixes = "KMGTPEZYRQ";
```

Ensures there's a NUL-byte that we can rely on in loops.

Reply:
Not having a NUL-byte was intentional by me as I use `ARRAYSIZE(unitPrefixes)` to get its length. If I have to pad a NUL byte to it (which wouldn't hurt much anyway), the length would have to change to `strlen(unitPrefixes)`. I just want to make sure you prefer using `strlen` to get its length.

Reply:
Technical P.S. I define this array as a `static` constant in an header file in order to aid compiler optimization. My testing showed that GCC could hard-code 'K' and 'Q' characters (the first and last of the unit prefixes) in certain branches if the array contents are exposed fully in one translation unit. That's why the array is not an `extern`.

Reply:
Ah, k. In that case, that's fine as is.